cirq/api/google/v2/program.proto

```diff
@@ -0,0 +1,154 @@
+syntax = "proto3";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "ProgramProto";
+option java_multiple_files = true;
+
+// A quantum program.
+message Program {
+  // The language in which the program is written.
+  Language language = 1;
+
+  // Programs can be specified by a circuit or a schedule.
+  oneof program {
+    // A circuit is an abstract representation as a series of moments, each
+    // moment having a set of gates that act on disjoint qubits. Circuits don't
+    // have absolute times for their operations (gates acting on qubits).
+    //
+    // When circuits are scheduled on hardware operations within a moment
+    // will all be scheduled to start at the same time, and the length of
+    // time associated with a moment will be equal to the duration of the
+    // longest operation in the moment. Subsequent moments will start as soon
+    // as the longest operation in the previous moment finishes.
```

Comment:
I would like to leave the scheduling strategy flexible here. We typically try to push gate schedules as late as possible, which by default would ignore the moment structure entirely. Even if we sync at the end of each moment, by default we would have all operations _ending_ together, instead of starting together. It's possible to schedule as described here, but I think it'd be best to leave it flexible. It might be good to have a place to send some "scheduling strategy" parameters that control how the scheduling happens; even if we don't define any such parameters yet, it would be nice to know where we _could_ add such parameters in the protos.